torch/csrc/distributed/c10d/Functional.cpp

```diff
@@ -182,34 +181,38 @@ std::vector<at::Tensor> reduce_scatter_tensor_coalesced(
 
 at::Tensor reduce_scatter_tensor(
     const at::Tensor& input,
-    std::string reduce_op,
+    const std::string& reduce_op,
     int64_t group_size,
-    std::string group_name) {
+    const std::string& group_name) {
   TORCH_CHECK(input.is_contiguous());
   std::vector<at::Tensor> inputs{input};
   return reduce_scatter_tensor_coalesced(
-      inputs, std::move(reduce_op), group_size, std::move(group_name))[0];
+      inputs, reduce_op, group_size, group_name)[0];
 }
 
 at::Tensor all_to_all_single(
     const at::Tensor& input,
-    std::vector<int64_t> output_split_sizes,
-    std::vector<int64_t> input_split_sizes,
+    const std::vector<int64_t>& output_split_sizes,
+    const std::vector<int64_t>& input_split_sizes,
     // NOLINTNEXTLINE(performance-unnecessary-value-param)
-    std::string group_name) {
+    const std::string& group_name) {
   TORCH_CHECK(input.is_contiguous());
   std::vector<int64_t> output_sizes = input.sizes().vec();
   output_sizes[0] = std::accumulate(
       output_split_sizes.begin(), output_split_sizes.end(), int64_t(0));
   auto output = input.new_empty(output_sizes);
 
   auto group = c10d::resolve_process_group(group_name);
+  // we could avoid this copy if we update all alltoall_base APIs
+  // to accept split sizes by const_ref
+  std::vector<int64_t> output_split_sizes_copy = output_split_sizes;
+  std::vector<int64_t> input_split_sizes_copy = input_split_sizes;
```

Comment:
in theory we might want to chagne `ProcessGroup.hpp::alltoall_base` to accept split sizes by const-ref too - I just chose to limit the blast radius of this PR